src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/SparseRangeInt32FrozenSet.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.Collections.Frozen
+{
+    /// <summary>Provides a frozen set to use when the value is an <see cref="int"/> and the default comparer is used and the items are in a contiguous range.</summary>
+    internal sealed class SparseRangeInt32FrozenSet : FrozenSetInternalBase<int, SparseRangeInt32FrozenSet.GSW>
+    {
+        private readonly int[] _items;
+        private readonly ulong[] _bits;
+        private readonly int _min;
+        private readonly int _max;
+
+        // assumes the items are in sorted order
+        internal SparseRangeInt32FrozenSet(int[] entries)
+            : base(EqualityComparer<int>.Default)
+        {
+            _items = entries;
+            _min = _items[0];
+            _max = _items[_items.Length - 1];
+
+            _bits = new ulong[((_max - _min + 1) + 63) / 64];
+
+            for (int i = 0; i < _items.Length; i++)
+            {
+                Calc(_items[i] - _min, out var slot, out var mask);
+                _bits[slot] |= mask;
+            }
+        }
+
+        private protected override int[] ItemsCore => _items;
+        private protected override Enumerator GetEnumeratorCore() => new Enumerator(_items);
+        private protected override int CountCore => _items.Length;
+
+        private protected override int FindItemIndex(int item)
+        {
+            if (item >= _min && item <= _max)
+            {
+                var bit = item - _min;
+
+                Calc(bit, out var slot, out var mask);
+                if ((_bits[slot] & mask) != 0UL)
+                {
+                    return bit;
+                }
+            }
+
+            return -1;
+        }
+
+        private static void Calc(int bit, out int slot, out ulong mask)
+        {
+            slot = (bit / 64);
+            mask = (1UL << (bit % 64));
+        }
```

Comment:
```suggestion
        private static void Calc(uint bit, out uint slot, out ulong mask)
        {
            slot = bit / 64;
            mask = 1UL << (int)(bit % 64);
        }
```
It's possible for the call-sites? If not keep the signature as is, but do the casts in the method body.
Codegen is [much simpler](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGAHKAJYA3bBhhNySJilbYANmAAUAgHYYGwARjQMIAV3Wr1uORG26DDPaZUBzBvjwBrAJRcenaj28MTZhgC8DIqa6gD0DKguANzuPg7OgcHkAKoAMgwAPJnBoQwApJEoLjFxDAC+ZWX8wqLixJLSsgrkinpGGlo6+urtar6m5j1WNvaOuK5lnvHcfupBeRGosV4z405JqRnZwUYuIVoFRaWr3JXU5UA).

Reply:
Done.